lib/saml_idp/service_provider.rb

```diff
@@ -19,6 +19,10 @@ def valid?
       attributes.present?
     end
 
+    def valid_identifier?
+      valid? && !identifier.nil?
```

Comment:
I.e. the only way to find a ServiceProvider from a Request is via the `Issuer` so if it is missing, the  ServiceProvider itself should be null. The reason it isn't is that the default `service_provider_finder` always returns the Default regardless of whether `Issuer` is defined.